src/include/sof/audio/audio_stream.h

```diff
@@ -60,6 +60,13 @@ struct audio_stream {
 	uint32_t rate;		/**< Number of data frames per second [Hz] */
 	uint16_t channels;	/**< Number of samples in each frame */
 
+	/**alignment limit of stream copy, alignment is the frame_align_shift-th
+	 * power of 2. And frame_align_shift should be set in component prepare
+	 * or param functions
+	 */
+	uint16_t frame_align_shift;
+	uint16_t frame_align;
```

Comment:
Every variable, constant and function in this PR is called "something_vague_align". If nothing else these two fields really need better names and a detailed doxygen to help understand why they are pre-computed (except when developers will forget) while the rest is not.